Storage/Storage/CoreData/IterativeStoreMigrator.swift

```diff
@@ -0,0 +1,220 @@
+                (try? NSMappingModel.inferredMappingModel(forSourceModel: modelFrom, destinationModel: modelTo)) else {
+                    return false
+            }
+
+            // Migrate the model to the next step
+            DDLogWarn("Attempting migration from \(modelNames[index]) to \(modelNames[index + 1])")
+            
+            guard migrateStore(at: sourceStore, storeType: storeType, fromModel: modelFrom, toModel: modelTo, with: migrateWithModel) == true else {
+                return false
+            }
+        }
+
+        return true
+    }
+}
+
+extension CoreDataIterativeMigrator {
+    private static func migrateStore(at url: URL, storeType: String, fromModel: NSManagedObjectModel, toModel: NSManagedObjectModel, with mappingModel: NSMappingModel) -> Bool {
+        // Build a temporary path to write the migrated store.
+        let fileManager = FileManager.default
+        let tempDestinationURL = url.deletingLastPathComponent().appendingPathComponent("migration").appendingPathComponent(url.lastPathComponent)
+        try? fileManager.removeItem(at: tempDestinationURL.deletingLastPathComponent())
+        try? fileManager.createDirectory(at: tempDestinationURL.deletingLastPathComponent(), withIntermediateDirectories: false, attributes: nil)
+
+        // Migrate from the source model to the target model using the mapping,
+        // and store the resulting data at the temporary path.
+        let migrator = NSMigrationManager(sourceModel: fromModel, destinationModel: toModel)
+        do {
+            try migrator.migrateStore(from: url, sourceType: storeType, options: nil, with: mappingModel, toDestinationURL: tempDestinationURL, destinationType: storeType, destinationOptions: nil)
+        } catch {
+            return false
+        }
+
+        // Move the original source store to a backup location.
+        let backupURL = url.deletingLastPathComponent().appendingPathComponent("backup")
+        try? fileManager.removeItem(at: backupURL)
+        try? fileManager.createDirectory(atPath: backupURL.path, withIntermediateDirectories: false, attributes: nil)
+        do {
+            let files = try fileManager.contentsOfDirectory(atPath: url.deletingLastPathComponent().path)
+            try files.forEach { (file) in
+                if file.hasPrefix(url.lastPathComponent) {
+                    let fullPath = url.deletingLastPathComponent().appendingPathComponent(file).path
+                    let toPath = URL(fileURLWithPath: backupURL.path).appendingPathComponent(file).path
+                    try fileManager.moveItem(atPath: fullPath, toPath: toPath)
+                }
+            }
+        } catch {
+            DDLogError("Error while moving original source store to a backup location: \(error)")
+
+            return false
+        }
+
+        // Copy migrated over the original files
```

Comment:
Similar to my previous comment, perhaps we could move this `Copy migrated over the original files` logic into a separate func?